feat(cxl-ui): add `cxl-dashboard-header` component

kertuilves opened the pull request:
https://app.clickup.com/t/861n0qhy4

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,114 @@
+                ${this.fastSkillsCourses}
+                <vaadin-icon icon="lumo:arrow-right"></vaadin-icon>
+              </vaadin-button>
+              ${!this.hasRoadmap
+                ? html`
+                  <vaadin-button class="roadmap">
+                    ${this.createRoadmap}
+                    <vaadin-icon icon="lumo:arrow-right"></vaadin-icon>
+                  </vaadin-button>`
+                : nothing
+              }
```
I can't find on Figma state for no roadmap but instead of showing button I'd display the same stats component just with `-` and button saying "Create roadmap" instead of "Edit"

kertuilves replied:
To me these are two completely different scenarios. One is when user has some ongoing roadmaps/courses and the second is when there isn't any. Could you explain your suggestion, I'm not sure if I got it right. 

<img width="706" alt="Screenshot 2023-07-31 at 16 31 01" src="https://github.com/conversionxl/aybolit/assets/119847199/059d780e-ce82-4bf5-80f1-3b18e7601b17">
<img width="693" alt="Screenshot 2023-07-31 at 16 31 08" src="https://github.com/conversionxl/aybolit/assets/119847199/7dcb4c3c-6c3a-4cad-a26a-6c0408920f0c">

pawelkmpt replied:
You're right @kertuilves. I didn't see then second design in Figma before

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,114 @@
+          <vaadin-icon icon="lumo:arrow-right"></vaadin-icon>
+        </vaadin-button>
+        <div class="progress">
+          <span class="progress-title">Completed 2 lessons of 6 lessons</span>
+          <vaadin-progress-bar value="0.334">Completed 2 lessons of 6 lessons</vaadin-progress-bar>
```
These will be dynamic, right?

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,114 @@
+            <vaadin-icon icon="lumo:angle-right"></vaadin-icon>
+          </vaadin-button>
+          <div>
+            <span class="subtitle">My dashboard</span>
+            <h1 class="title">Hello ${this.name}!</h1>
```
I suggest making these properties with default values:
* `title` -> 'Hello'
* `subtitle` -> 'My dashboard'

We need more flexibility and changing the header should not require editing a component.

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,114 @@
+      <div class="last-course">
+        <vaadin-button>
+          <div>
+            <span class="subtitle">Continue where you left off</span>
```
```suggestion
            <span class="subtitle">{this.lastCourseSubtitle}</span>
```

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,114 @@
+        <section class="content">
+          <div>
+            ${this.renderLastCourse()}
+            <div class="courses">
+              <vaadin-button>
+                ${this.allCourses}
+                <vaadin-icon icon="lumo:arrow-right"></vaadin-icon>
+              </vaadin-button>
+              <vaadin-button>
+                ${this.fastSkillsCourses}
+                <vaadin-icon icon="lumo:arrow-right"></vaadin-icon>
+              </vaadin-button>
```
In design these are indeed courses but to make this section more flexible I suggest to rename it to `cta` and buttons to `cta-1`, `cta-2` and possibility to pass URLs too.

pawelkmpt reviewed (changes requested):
@kertuilves this is going into the good direction 👍  I see some styles are missing and not everything is completed yet but it's a good base for further work. Please check my suggestions

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,138 @@
+
+  @property({ type: String }) name = '';
+
+  @property({ type: String }) notificationCount = '6';
```
Should be `0` by default and also `Number` type

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,138 @@
+
+  @property({ type: String }) notificationCount = '6';
+
+  @property({ type: String }) updateTitle = "What's new in CXL";
```
Keep it consistent with `notificationCount`

```suggestion
  @property({ type: String }) notificationsTitle = "What's new in CXL";
```

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,138 @@
+
+  @property({ type: String }) updateTitle = "What's new in CXL";
+
+  @property({ type: String }) lastCourseTitle = 'Advanced experimentation analysis';
```
```suggestion
  @property({ type: String }) lastCourseTitle = '';
```

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,138 @@
+
+  @property({ type: String }) lastCourseTitle = 'Advanced experimentation analysis';
+
+  @property({ type: String }) lastCourseSubitle = 'Continue where you left off';
```
```suggestion
  @property({ type: String }) lastCourseSubtitle = 'Continue where you left off';
```

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,138 @@
+
+  @property({ type: String }) lastCourseSubitle = 'Continue where you left off';
+
+  @property({ type: String }) progressbarTitle = 'Completed 2 lessons of 6 lessons';
```
it can't be hardcoded

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,138 @@
+
+  @property({ type: String }) progressbarTitle = 'Completed 2 lessons of 6 lessons';
+
+  @property({ type: String }) progressbarValue = '0.34';
```
```suggestion
  @property({ type: Number }) progressbarValue = '0';
```

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,138 @@
+
+  @property({ type: String }) ctaLink = '';
+
+  @property({ type: String }) cta1 = 'Browse fast skills';
```
it can't be hardcoded

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,138 @@
+
+  @property({ type: String }) cta1Link = '';
+
+  @property({ type: String }) cta2 = 'Create your personal learning roadmap';
```
As I understand  this is the second button and roadmap button is the 3rd, so this can't be used as such CTA.

pawelkmpt reviewed (changes requested):
It doesn't look good :(

![Screenshot 2023-08-09 at 11 39 32](https://github.com/conversionxl/aybolit/assets/5165721/0377d4fe-e553-495b-951a-f5f5ea14a3de)
![Screenshot 2023-08-09 at 11 39 38](https://github.com/conversionxl/aybolit/assets/5165721/1acd433b-c4af-4813-82d9-30a4a5cc9c81)

kertuilves commented:
> It doesn't look good :(
> 
> ![Screenshot 2023-08-09 at 11 39 32](https://user-images.githubusercontent.com/5165721/259365653-0377d4fe-e553-495b-951a-f5f5ea14a3de.png) ![Screenshot 2023-08-09 at 11 39 38](https://user-images.githubusercontent.com/5165721/259365662-1acd433b-c4af-4813-82d9-30a4a5cc9c81.png)

@pawelkmpt Can you provide more information like browser and version? I tested it in my browsers (Safari, Chrome and Firefox) and all is well there.

pawelkmpt reviewed (changes requested):
Code is good now, styles too.

There are 2 items which are not clickable but are suppose to. Storybook thing or need for component tweak?
* "CONTINUE WHERE YOU LEFT OFF"
* Edit roadmap

kertuilves commented:
> Code is good now, styles too.
> 
> There are 2 items which are not clickable but are suppose to. Storybook thing or need for component tweak?
> 
> * "CONTINUE WHERE YOU LEFT OFF"
> * Edit roadmap

* I changed this Last Course part to `<a>` and added link property. 
* I added onclick links for `<vaadin-button>`'s. I thought this part would be done when actually including the component.

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,139 @@
+          </a>
+          <div class="progress">
+            <span class="progress-title"
+              >Completed ${this.lessonsCompleted} lessons of ${this.lessonsTotal} lessons</span
```
```suggestion
              >Completed ${this.lessonsCompleted} of ${this.lessonsTotal} lessons</span
```

pawelkmpt commented on packages/cxl-ui/src/components/cxl-dashboard-header.js:
```diff
@@ -0,0 +1,139 @@
+              >Completed ${this.lessonsCompleted} lessons of ${this.lessonsTotal} lessons</span
+            >
+            <vaadin-progress-bar value="${this.progress}"
+              >Completed ${this.lessonsCompleted} lessons of ${this.lessonsTotal}
+              lessons</vaadin-progress-bar
```
```suggestion
              >Completed ${this.lessonsCompleted} of ${this.lessonsTotal}
              lessons</vaadin-progress-bar
```

pawelkmpt reviewed (changes requested):
Small tweak and we can merge

pawelkmpt commented:
We have a conflict @kertuilves
